x-pack/plugins/observability_solution/apm/public/components/app/mobile/service_overview/geo_map/embedded_map.tsx

```diff
@@ -34,137 +29,41 @@ function EmbeddedMapComponent({
-            },
-          }),
-        });
-        return;
-      }
+  const { maps } = useKibana<ApmPluginStartDeps>().services;
 
-      const input: MapEmbeddableInput = {
-        attributes: { title: '' },
-        id: uuidv4(),
-        title: i18n.translate('xpack.apm.serviceOverview.embeddedMap.input.title', {
-          defaultMessage: 'Latency by country',
-        }),
-        filters,
-        viewMode: ViewMode.VIEW,
-        mapCenter: { lat: 20.43425, lon: 0, zoom: 1.25 },
-        isLayerTOCOpen: false,
-        query: {
-          query: kuery,
-          language: 'kuery',
-        },
-        timeRange: {
-          from: start,
-          to: end,
-        },
-        hideFilterActions: true,
-      };
-
-      const embeddableObject = await factory.create(input);
-
-      setEmbeddable(embeddableObject);
-    }
-
-    setupEmbeddable();
-    // Set up exactly once after the component mounts
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
-
-  // We can only render after embeddable has already initialized
-  useEffect(() => {
-    if (embeddableRoot.current && embeddable) {
-      embeddable.render(embeddableRoot.current);
-    }
-  }, [embeddable, embeddableRoot]);
-
-  useEffect(() => {
-    const setLayerList = async () => {
-      if (embeddable && !isErrorEmbeddable(embeddable) && dataView?.id) {
-        const layerList = await getLayerList({
-          selectedMap,
-          maps,
-          dataViewId: dataView.id,
-        });
-        await Promise.all([embeddable.setLayerList(layerList), embeddable.reload()]);
-      }
-    };
-
-    setLayerList();
-  }, [embeddable, selectedMap, maps, dataView]);
-
-  useEffect(() => {
-    if (embeddable) {
-      embeddable.updateInput({
-        filters,
-        query: {
-          query: kuery,
-          language: 'kuery',
-        },
-        timeRange: {
-          from: start,
-          to: end,
-        },
-      });
-    }
-  }, [start, end, kuery, filters, embeddable, selectedMap]);
+  const layerList = dataView?.id
+    ? getLayerList({
+        selectedMap,
+        maps,
+        dataViewId: dataView?.id,
+      })
+    : [];
 
   return (
-    <>
-      {error && (
-        <EuiText size="s">
-          <p>
-            {i18n.translate('xpack.apm.serviceOverview.embeddedMap.error', {
-              defaultMessage: 'Could not load map',
-            })}
-          </p>
-        </EuiText>
-      )}
-      {!error && (
-        <div
-          data-test-subj="serviceOverviewEmbeddedMap"
-          css={css`
-            width: 100%;
-            height: 500px;
-            display: flex;
-            flex: 1 1 100%;
-            z-index: 1;
-            min-height: 0;
-          `}
-          ref={embeddableRoot}
-        />
-      )}
-    </>
+    <div
+      data-test-subj="serviceOverviewEmbeddedMap"
+      style={{ width: '100%', height: '500px', display: 'flex', flex: '1 1 100%', zIndex: 1 }}
```

Comment:
Using `style` instead of the emotion `css` method resolves the issue that also affects the main branch where the height of this div may change to 0 when the filters change.


https://github.com/elastic/kibana/assets/1638483/286060a6-bcd0-4527-9a29-5de6ec46e650


https://github.com/elastic/kibana/assets/1638483/542ad957-31b0-4acc-a3dc-4a984900f269